examples/99-advanced/planets.py

```diff
@@ -0,0 +1,119 @@
+
+
+# Light of the sun in the +x axis direction.
+
+light = pyvista.Light()
+light.set_direction_angle(30, -20)
+
+# Load planet
+
+# https://tamaskis.github.io/files/Visualizing_Celestial_Bodies_in_3D.pdf
+# Mercury's radius is 2439.0 km
+mercury = examples.planets.load_mercury(radius=2439.0, lat_resolution=150, lon_resolution=300)
+# Venus's radius is 6052.0 km
+venus = examples.planets.load_venus(radius=6052.0, lat_resolution=150, lon_resolution=300)
+# Mars's radius is 3397.2 km
+mars = examples.planets.load_mars(radius=3397.2, lat_resolution=150, lon_resolution=300)
+# Jupiter's radius is 71492.0 km
+jupiter = examples.planets.load_jupiter(radius=71492.0, lat_resolution=150, lon_resolution=300)
+# Saturn's radius is 60268.0 km
+saturn = examples.planets.load_saturn(radius=60268.0, lat_resolution=150, lon_resolution=300)
+# Saturn's rings range from 7000.0 km to 80000.0 km from the surface of the planet
+inner = 60268.0 + 7000.0
+outer = 60268.0 + 80000.0
+saturn_ring_alpha = examples.planets.load_saturn_ring_alpha(inner=inner, outer=outer, c_res=50)
+# Uranus's radius is 25559.0 km
+uranus = examples.planets.load_uranus(radius=25559.0, lat_resolution=150, lon_resolution=300)
+# Neptune's radius is 24764.0 km
+neptune = examples.planets.load_neptune(radius=24764.0, lat_resolution=150, lon_resolution=300)
+
+# Move planet position (Numbers have no meaning. The planets are laid out for
+# easy viewing.).
+
+mercury.translate((0.0, 0.0, 0.0), inplace=True)
+venus.translate((-15000.0, 0.0, 0.0), inplace=True)
+mars.translate((-45000.0, 0.0, 0.0), inplace=True)
+jupiter.translate((-150000.0, 0.0, 0.0), inplace=True)
+saturn.translate((-400000.0, 0.0, 0.0), inplace=True)
+saturn_ring_alpha.translate((-400000.0, 0.0, 0.0), inplace=True)
+uranus.translate((-600000.0, 0.0, 0.0), inplace=True)
+neptune.translate((-700000.0, 0.0, 0.0), inplace=True)
+
+# Add planets to Plotter.
+
+plotter = pyvista.Plotter(lighting="none")
+cubemap = examples.download_cubemap_space_16k()
+_ = plotter.add_actor(cubemap.to_skybox())
+plotter.set_environment_texture(cubemap, True)
+plotter.add_light(light)
+plotter.add_mesh(mercury)
+plotter.add_mesh(venus)
+plotter.add_mesh(mars)
+plotter.add_mesh(jupiter)
+plotter.add_mesh(saturn)
+plotter.add_mesh(saturn_ring_alpha)
+plotter.add_mesh(uranus)
+plotter.add_mesh(neptune)
+plotter.camera.zoom(1.5)
+plotter.show()
+
+###############################################################################
+# `Textures in this pack are based on NASA <https://www.solarsystemscope.com/textures/https://www.solarsystemscope.com/textures/>`_
+# elevation and imagery data. Colors and shades of the textures are tuned
+# according to true-color photos made by Messenger, Viking and Cassini
+# spacecrafts, and, of course, the Hubble Space Telescope.
+
+plotter = pyvista.Plotter(shape=(3, 2))
+plotter.subplot(0, 0)
+plotter.add_text("Mercury")
+plotter.add_mesh(pyvista.read(examples.planets.download_mercury_jpg(load=False)), rgb=True)
```

Comment:
Why do we need a combo of `pv.read()` + `load=False` in these calls? Couldn't we just use `load=True`?

Reply:
Your right. Thanks.
```suggestion
plotter.add_mesh(examples.planets.download_mercury_jpg(), rgb=True)
```

Reply:
I will revert this because that vtk cannot load jpg file.
```bash
WARNING:root:ERROR: In vtkPNGReader.cxx, line 131
(nullptr): Unknown file type! Not a PNG file!
```

Reply:
In my local.
```python
   2830     mesh = wrap(mesh)
   2831     if not is_pyvista_dataset(mesh):
-> 2832         raise TypeError(
   2833             f'Object type ({type(mesh)}) not supported for plotting in PyVista.'
   2834         )
   2835 elif isinstance(mesh, pyvista.PointSet):
   2836     # cast to PointSet to PolyData
   2837     mesh = mesh.cast_to_polydata(deep=False)
```

Reply:
Revert in 5900e2966b55ae5795db90e255d4633198b5b5e2 .

Reply:
See https://github.com/pyvista/pyvista/issues/3301

Reply:
These methods are loading as a `pyvista.Texture` (not a mesh and thus not supported by `add_mesh`).

```py
pyvista.read(examples.planets.download_mercury_jpg(load=False))
```

is doing a few things:

1. `examples.planets.download_mercury_jpg(load=False)` returns file name/path
2. `pyvista.read()` loads the filename, in this case image data, as `vtkImageData` or `pyvista.UniformGrid` (a mesh type)

Switching this to just `examples.planets.download_mercury_jpg()` loads the file as a `pyvista.Texture`

Reply:
Sorry, it was my misunderstanding. Added 6eef984ed50f15a3bb3b3ff77c7d4332435bc7cf.

Reply:
Thanks @banesullivan I wonder if we should rename all these methods to `download_mercury_texture()` etc. then.

Reply:
Yeah, I think we may need to drive home that these load as `Texture` objects rather than meshes. Renaming seems appropriate to me (for another PR)